os/drivers/video/video_null_driver.c

```diff
@@ -237,6 +262,87 @@ struct v4l2_mapping_s ctrl_mapping[] = {
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+
+/****************************************************************************
+ * Name: video_null_set_worker_delay
+ *
+ * Description:
+ *   This function calculates the delay requred for the worker thread
+ *   to invoke a callback based on the resolution chosen and the
+ *   video frames-per-second required.
+ *
+ * Input Paramaters:
+ *   priv - pointer to null_priv_t object holding the frame
+ *   resolution attributes.
+ *
+ * Returned Values:
+ *   OK on success, appropriate error code otherwise.
+ ****************************************************************************/
+static int video_null_set_worker_delay(FAR null_priv_t *priv)
```

Comment:
This function calculates the delay as follows - 
1. Based on the frame resolution, a standard delay has been calculated assuming a required transfer at 30fps.